File: backend/src/main/java/com/documed/backend/prescription/Prescription.java
package com.documed.backend.prescription;

import com.documed.backend.visits.Visit;
import java.util.Date;
import lombok.Data;
import lombok.NonNull;

@Data
public class Prescription {
  private int id;
  private int accessCode;
  private String description;
  private final Date date;
  @NonNull private Date expirationDate;
  private int pesel;
  private String passportNumber;
  @NonNull private Visit visit;
}

File: frontend/src/shared/types/Prescription.ts
import { PrescribedMedicine } from './PrescribedMedicine';
import { VisitLite } from './Visit';

export interface Prescription extends PrescriptionLite {
  visit: VisitLite;
}

export interface PrescriptionLite {
  medicines: PrescribedMedicine[];
  id: number;
  accessCode: number | null;
  description: string | null;
  date: Date | null;
  expirationDate: Date;
  pesel: number | null;
  passportNumber: string | null;
}
